icechunk/src/repository.rs

```diff
@@ -382,17 +379,13 @@ impl Repository {
     }
 
     #[instrument(skip_all)]
-    pub async fn set_default_commit_metadata(
-        &self,
-        metadata: Option<SnapshotProperties>,
-    ) {
-        let mut guard = self.default_commit_metadata.write().await;
-        *guard = metadata;
+    pub fn set_default_commit_metadata(&mut self, metadata: SnapshotProperties) {
+        self.default_commit_metadata = metadata;
     }
 
     #[instrument(skip_all)]
-    pub async fn default_commit_metadata(&self) -> Option<SnapshotProperties> {
-        self.default_commit_metadata.read().await.clone()
+    pub fn default_commit_metadata(&self) -> SnapshotProperties {
```

Comment:
I think in here I'd prefer to return `&SnapshotProperties`. That way the caller can clone it only if they need to.